### PR TITLE
Fix some pylint warnings

### PR DESCRIPTION
**Pull request by dgreen-arm**
This PR fixes a few warnings that the check_python_files component of all.sh was complaining about

### CODE REVIEW

File: tests/scripts/check-test-cases.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
 import sys
 
 class Results:
+    """Stores file and line information about errors
```
Comment (gilles-peskine-arm) [inline tests/scripts/check-test-cases.py]: Nitpick: [PEP 257](https://www.python.org/dev/peps/pep-0257) says to write ”Store …“, not “Stores …”, and to end with a period. PEP 257 also says a one-line docstring “should really fit on one line”, and a multi-line docstring should start with a one-line summary.

File: tests/scripts/mbedtls_test.py [hunk 2]
```diff
@@ -310,7 +310,10 @@ def run_test(self, name, function_id, dependencies, args):
 
         param_bytes, length = self.test_vector_to_bytes(function_id,
                                                         dependencies, args)
-        self.send_kv(''.join('{:02x}'.format(x) for x in length), ''.join('{:02x}'.format(x) for x in param_bytes))
+        self.send_kv(
+            ''.join('{:02x}'.format(x) for x in length),
+            ''.join('{:02x}'.format(x) for x in param_bytes)
+        )
```
Comment (gilles-peskine-arm) [inline tests/scripts/mbedtls_test.py]: How about
```
params = length + param_bytes
self.send_kv('.join('{:02x}'.format(x) for x in params))
```

Comment (dgreen-arm) [reply tests/scripts/mbedtls_test.py]: Wouldn't that send one string, rather than two separate strings?

Comment (gilles-peskine-arm) [reply tests/scripts/mbedtls_test.py]: Oh right. Never mind.

File: tests/scripts/check-test-cases.py (hunk 1 above)
Comment (dgreen-arm) [reply tests/scripts/check-test-cases.py]: I've updated this.

File: tests/scripts/check-test-cases.py [hunk 3]
```diff
@@ -41,6 +42,7 @@ def warning(self, file_name, line_number, fmt, *args):
         self.warnings += 1
 
 def collect_test_directories():
+    """Gets the relative path for the TLS and Crypto test directories."""
```
Comment (Patater) [inline tests/scripts/check-test-cases.py]: Get

File: tests/scripts/check-test-cases.py [hunk 4]
```diff
@@ -55,6 +57,7 @@ def collect_test_directories():
     return directories
 
 def check_description(results, seen, file_name, line_number, description):
+    """Checks test case descriptions for errors."""
```
Comment (Patater) [inline tests/scripts/check-test-cases.py]: Check

**Comment by Patater** [comment]
Mbed OS tests expected to be failing until https://github.com/ARMmbed/mbed-os/pull/12150 is merged

**Pull request merged**